// jscpd-badge-reporter/build/module/index.js
import badgen from 'badgen';
import { writeFileSync } from 'fs';
import { ensureDirSync } from 'fs-extra';
import { join } from 'path';
export default class {
    constructor(options) {
        this.options = options;
    }
    report(...args) {
        const [, statistic] = args;
        const badgeOptions = this.options.reportersOptions ? this.options.reportersOptions.badge || {} : {};
        const badge = badgen({
            color: this.getColor(statistic),
            status: this.getStatus(statistic),
            subject: 'Copy/Paste',
            ...badgeOptions
        });
        ensureDirSync(this.options.output);
        writeFileSync(badgeOptions.path ? badgeOptions.path : join(this.options.output, 'jscpd-badge.svg'), badge);
    }
    attach() {
    }
    getStatus(statistic) {
        return statistic ? statistic.total.percentage + '%' : 'N/A';
    }
    getColor(statistic) {
        if (!statistic || !statistic.hasOwnProperty('threshold')) {
            return 'grey';
        }
        return statistic.total.percentage < statistic.threshold ? 'green' : 'red';
    }
}
