// jscpd-badge-reporter/build/main/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const badgen_1 = __importDefault(require("badgen"));
const fs_1 = require("fs");
const fs_extra_1 = require("fs-extra");
const path_1 = require("path");
class default_1 {
    constructor(options) {
        this.options = options;
    }
    report(...args) {
        const [, statistic] = args;
        const badgeOptions = this.options.reportersOptions ? this.options.reportersOptions.badge || {} : {};
        const badge = badgen_1.default(Object.assign({ color: this.getColor(statistic), status: this.getStatus(statistic), subject: 'Copy/Paste' }, badgeOptions));
        fs_extra_1.ensureDirSync(this.options.output);
        fs_1.writeFileSync(badgeOptions.path ? badgeOptions.path : path_1.join(this.options.output, 'jscpd-badge.svg'), badge);
    }
    attach() {
    }
    getStatus(statistic) {
        return statistic ? statistic.total.percentage + '%' : 'N/A';
    }
    getColor(statistic) {
        if (!statistic || !statistic.hasOwnProperty('threshold')) {
            return 'grey';
        }
        return statistic.total.percentage < statistic.threshold ? 'green' : 'red';
    }
}
exports.default = default_1;
